src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs

```diff
@@ -165,40 +184,74 @@ private ResolutionDetails<T> ResolveValue<T>(string flagKey, T defaultValue, Eva
                     // convert the EvaluationContext object into something the JsonLogic evaluator can work with
                     dynamic contextObj = (object)ConvertToDynamicObject(targetingContext);
 
-                    variant = (string)_evaluator.Apply(rule, contextObj);
-
+                    // convery whatever is returned to a sring to try to use is an a index to Variants
+                    var ruleResult = _evaluator.Apply(rule, contextObj);
+                    if (ruleResult is bool)
+                    {
+                        // if this was a bool, convert from "True" to "true" to match JSON
+                        variant = Convert.ToString(ruleResult).ToLower();
+                    }
+                    else
+                    {
+                        // convert whatever is returned to a string to support shorthand
+                        variant = Convert.ToString(ruleResult);
+                    }
                 }
 
-
                 // using the returned variant, go through the available variants and take the correct value if it exists
-                if (variant != null && flagConfiguration.Variants.TryGetValue(variant, out var foundVariantValue))
+                if (variant == null)
                 {
-                    if (foundVariantValue is Int64)
-                    {
-                        checked
-                        {
-                            foundVariantValue = Convert.ToInt32(foundVariantValue);
-                        }
-                    }
-                    else if (foundVariantValue is JObject value)
+                    // if variant is null, revert to default
```

Comment:
nit - if a variant is null, should we throw it instead of attempting to derive a default variant?  For example, in Java, we yeld a type mismatch error [1] 

[1] - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java#L197-L201

Reply:
I could be confused, but I don't think that line in the Java implementation is doing the same thing. The line in Java appears to be throwing _if the variant retrieved by `resolvedVariant` is null_. That's not the same thing as `resolvedVariant` being null itself. If the resolved variant itself is null, it means our targeting has essentially fallen through, and we will fall back to the default - that's what this line is doing. The Java impl does this as well (the e2e suite contains a specific test flag for this (`targeting-null-variant-flag`).